packages/browser-wallet/src/background/recovery.ts

```diff
@@ -165,12 +220,34 @@ async function performRecovery({ providers, ...recoveryInputs }: Payload) {
     }
 }
 
-export const recoveryHandler: ExtensionMessageHandler = (msg) => {
-    const respond = async (result: RecoveryBackgroundResponse) => {
-        await openWindow();
-        bgMessageHandler.sendInternalMessage(InternalMessageType.RecoveryFinished, result);
-    };
-    performRecovery(msg.payload)
-        .then((added) => respond({ status: BackgroundResponseStatus.Success, added }))
-        .catch((e) => respond({ status: BackgroundResponseStatus.Error, reason: e.toString() }));
-};
+export function startRecovery() {
+    chrome.alarms.create(RECOVERY_ALARM_NAME, { delayInMinutes: 0 });
+}
+
+export async function setupRecoveryHandler() {
+    chrome.alarms.onAlarm.addListener(async (alarm) => {
+        if (alarm.name !== RECOVERY_ALARM_NAME) {
+            return;
+        }
+        const isRecovering = await sessionIsRecovering.get();
+        if (isRecovering) {
+            // We use a lock to ensure only 1 recovery instance runs
+            navigator.locks.request(RECOVERY_LOCK, { ifAvailable: true }, (lock) => {
+                if (!lock) {
+                    // The lock was not granted - get out fast.
+                    return Promise.resolve();
+                }
+                chrome.alarms.create(RECOVERY_ALARM_NAME, { delayInMinutes: 5.1, periodInMinutes: 1 });
```

Comment:
Should `delayInMinutes` not be `< 5` minutes?

Reply:
The way I understand it now is that the alarm can be used to revive the SW, not keep it alive. So I want it to happen right after the SW dies, not before.